ui/src/app/settings/views/Scripts/ScriptsUpload.js

```diff
@@ -0,0 +1,189 @@
+};
+
+const rejectStyle = {
+  borderColor: "#ff1744"
+};
+
+const ScriptsUpload = ({ type }) => {
+  const MAX_SIZE_BYTES = 2000000; // 2MB
+  const hasErrors = useSelector(selectors.scripts.hasErrors);
+  const errors = useSelector(selectors.scripts.errors);
+  const saved = useSelector(selectors.scripts.saved);
+  const [savedScript, setSavedScript] = useState();
+  const [script, setScript] = useState();
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (hasErrors) {
+      Object.keys(errors).forEach(key => {
+        dispatch(
+          messages.add(
+            `Error uploading ${savedScript}: ${errors[key]}`,
+            "negative"
+          )
+        );
+      });
+      dispatch(actions.scripts.cleanup());
+    }
+  }, [savedScript, hasErrors, errors, dispatch]);
+
+  const onDrop = useCallback(
+    (acceptedFiles, rejectedFiles) => {
+      if (acceptedFiles.length > 1 || rejectedFiles.length > 1) {
+        dispatch(
+          messages.add(`Only a single file may be uploaded.`, "caution")
+        );
+        return;
+      }
+
+      if (rejectedFiles.length > 0 && rejectedFiles[0].size > MAX_SIZE_BYTES) {
+        dispatch(
+          messages.add(
+            `File size must be ${MAX_SIZE_BYTES} bytes, or fewer.`,
+            "caution"
+          )
+        );
+        return;
+      }
+
+      if (rejectedFiles.length > 0) {
+        dispatch(
+          messages.add(`Invalid filetype, please try again.`, "negative")
+        );
+        return;
+      }
+
+      const acceptedFile = acceptedFiles[0];
+      const scriptName = pathParse(acceptedFile.path).name;
+      const reader = new FileReader();
+      reader.onabort = () => console.error("file reading was aborted");
+      reader.onerror = () => console.error("file reading has failed");
+      reader.onload = () => {
+        const binaryStr = reader.result;
+        setScript({ name: scriptName, script: binaryStr });
+      };
+
+      reader.readAsBinaryString(acceptedFile);
+    },
+    [dispatch]
+  );
+
+  const {
+    acceptedFiles,
+    getRootProps,
+    getInputProps,
+    isDragActive,
+    isDragAccept,
+    isDragReject
+  } = useDropzone({
+    onDrop,
+    accept: "text/*",
+    maxSize: MAX_SIZE_BYTES,
+    multiple: false
+  });
+
+  const style = useMemo(
+    () => ({
+      ...baseStyle,
+      ...(isDragActive ? activeStyle : {}),
+      ...(isDragAccept ? acceptStyle : {}),
+      ...(isDragReject ? rejectStyle : {})
+    }),
+    [isDragAccept, isDragActive, isDragReject]
+  );
+
+  useEffect(() => {
+    if (saved) {
+      dispatch(actions.scripts.cleanup());
+      dispatch(
+        messages.add(`${savedScript} uploaded successfully.`, "information")
+      );
+      setSavedScript();
+    }
+  }, [dispatch, saved, savedScript]);
+
+  if (saved) {
+    // The script was successfully uploaded so redirect to the scripts list.
+    return <Redirect to={`/scripts/${type}`} />;
+  }
+
+  return (
+    <Card>
```

Comment:
It might be nice to update the `FormCard` to handle the case where the heading is on its own row and use that here.